backend/eventloop/db_reconciler.go

```diff
@@ -385,6 +385,11 @@ func cleanOrphanedEntriesfromTable_ACTDM_GitOpsDeploymentSyncRun(ctx context.Con
 		return
 	}
 
+	if syncOperationDb.Application_id == "" {
```

Comment:
Rather than exiting here, we likely want to just skip the parts of this function that require this field to be non-empty (correct me if I'm wrong!)

For example, when Application_id == "", we probably still want to do most of the rest of the function:
- Update DesiredState of SyncOperation
- Create an Operation pointing to the SyncOperation
- Delete the SyncOperation

I'm actually slightly surprised the E2E tests passed with this change :thinking:, I presume because we aren't testing SyncOperation where application_id empty...

Reply:
Yeah. Since it's not possible to create operations without accessing the GitOpsEngineInstance, we can directly delete the resources without creating an operation.